client/fingerprint_manager.go

```diff
@@ -282,20 +283,69 @@ func (fm *FingerprintManager) watchDriverFingerprint(fpChan <-chan *drivers.Fing
 		case <-fm.shutdownCh:
 			cancel()
 			return
-		case fp := <-fpChan:
-			di := &structs.DriverInfo{
-				Attributes:        fp.Attributes,
-				Detected:          fp.Health != drivers.HealthStateUndetected,
-				Healthy:           fp.Health == drivers.HealthStateHealthy,
-				HealthDescription: fp.HealthDescription,
-				UpdateTime:        time.Now(),
-			}
-			fm.nodeLock.Lock()
-			n := fm.updateNodeFromDriver(name, di)
-			if n != nil {
-				fm.node = n
+		case fp, ok := <-fpChan:
```

Comment:
Thereis too much nesting here. Make fp and okay variables and then handle outside the select

Reply:
I reduced the nesting a bit here but I didn't seperate out the logic